autogluon/utils/tabular/ml/models/lr/lr_model.py

```diff
@@ -0,0 +1,204 @@
+import logging
+import re
+
+import numpy as np
+from pandas import DataFrame
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.impute import SimpleImputer
+from sklearn.pipeline import FeatureUnion, Pipeline
+from sklearn.preprocessing import StandardScaler, QuantileTransformer
+
+from autogluon.utils.tabular.ml.constants import BINARY, REGRESSION
+from autogluon.utils.tabular.ml.models.abstract.abstract_model import AbstractModel
+from autogluon.utils.tabular.ml.models.lr.hyperparameters.parameters import get_param_baseline, get_model_params, get_default_params, INCLUDE, IGNORE, ONLY
+from autogluon.utils.tabular.ml.models.lr.hyperparameters.searchspaces import get_default_searchspace
+from autogluon.utils.tabular.ml.models.lr.lr_preprocessing_utils import NlpDataPreprocessor, OheFeaturesGenerator, NumericDataPreprocessor
+
+logger = logging.getLogger(__name__)
+
+
+class LinearModel(AbstractModel):
+
+    def __init__(self, path: str, name: str, problem_type: str, objective_func, hyperparameters=None, features=None,
+                 feature_types_metadata=None, debug=0, **kwargs):
+        self.model_class, self.penalty, self.handle_text = get_model_params(problem_type, hyperparameters)
+        name = name + '-' + self.model_class.__name__
+        name = name + '-text_' + self.handle_text
```

Comment:
Lets avoid altering the name based on the hyperparameters. If the user wants to know more about a given linear learner, they can look at the info metadata.

Proposing to just name it as LinearModel_0, LinearModel_1, etc. (Same name as passed in init)

Reply:
Updated